src/App.jsx

```diff
@@ -14,33 +16,92 @@ export const goodsFromServer = [
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const SORT_ALPHABETICALLY = 'Sort alphabetically';
+  const SORT_BY_LENGTH = 'Sort by length';
+  const REVERSE = 'Reverse';
+  const RESET = 'Reset';
+  const [sortField, setSortField] = useState(null);
+  const [reverse, setReverse] = useState(false);
+  let visibleGoods = [...goodsFromServer];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  switch (sortField) {
+    case SORT_ALPHABETICALLY:
+      visibleGoods = [...goodsFromServer].sort((a, b) => (a > b ? 1 : -1));
+      break;
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+    case SORT_BY_LENGTH:
+      visibleGoods = [...goodsFromServer].sort(
+        (good1, good2) => good1.length - good2.length,
+      );
+      break;
+    default:
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (reverse) {
+    visibleGoods = [...visibleGoods].reverse();
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          onClick={() => setSortField(SORT_ALPHABETICALLY)}
+          className={classNames('button', 'is-info', {
+            'is-light': sortField !== SORT_ALPHABETICALLY,
+          })}
+        >
+          {SORT_ALPHABETICALLY}
+        </button>
+        <button
+          type="button"
+          onClick={() => setSortField(SORT_BY_LENGTH)}
+          className={classNames('button', 'is-success', {
+            'is-light': sortField !== SORT_BY_LENGTH,
+          })}
+        >
+          {SORT_BY_LENGTH}
+        </button>
+        <button
+          type="button"
+          onClick={() => {
+            setReverse(!reverse);
+            if (sortField === REVERSE) {
+              setSortField(null);
+            } else if (sortField === null) {
+              setSortField(REVERSE);
+            }
+          }}
+          className={classNames('button', 'is-warning', {
+            'is-light': !reverse,
+          })}
+        >
+          {REVERSE}
+        </button>
+        {sortField && (
+          <button
+            type="button"
+            className={classNames('button', 'is-danger', {
+              'is-light': !sortField,
+            })}
+            onClick={() => {
+              setSortField(null);
+              setReverse(false);
+            }}
+          >
+            {RESET}
+          </button>
+        )}
+      </div>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+      <ul className="GoodsList">
+        {visibleGoods.map(good => (
+          <li data-cy="Good" key={good}>
```

Comment:
The `key` prop should be unique for each item in the list. Consider using the index of the item in the array or another unique identifier to ensure React can efficiently update the list.